kadi/commands/states.py

```diff
@@ -51,6 +52,17 @@
                       'vid_board')
 
 
+@contextlib.contextmanager
+def disable_grating_move_duration():
```

Comment:
It seems that this is used only in testing. Would this be required elsewhere? Can the reference data in testing be changed instead of using this function?

I am assuming this does not matter, but it changes a "static" class class member. There is not much threaded code, I suppose, but something like this could impact an instance running in another thread, right?

Reply:
My impression is that it is only used in testing here but the intent is that this would be a public method to allow a user to turn off the grating "move" times as desired.  And that this code is probably not thread safe.

Reply:
I originally wrote this just for testing, but decided it could just as well go as a public method in case someone discovers they need to have the legacy version without the move states. A rule of thumb from astropy is that whenever you make a change that breaks a bunch of your own unit tests, then beware!

About being thread safe, good point. I think there is minimal risk of a problem here, but for the sake of argument how might this functionality be accomplished in a thread safe way?

Reply:
It is not trivial to guarantee thread safety, and I know it is not warranted in this case because we don't use threads in this case. I would have to think.

A class member is basically a global variable, and global variables lead to unexpected behavior, because you as a user are not thinking about them. That is why this is my knee-jerk reaction.

It is a bit better if `apply_move_duration` is a data member of an instance, and not of the class, because in this case any thread-related issues would happen if you use the same instance at the same time from two threads, which can be expected. However, you always use the module-level functions, so I can't really propose a simple alternative.

Reply:
Understood. With that I propose you just approve and we go ahead with this. Looks like @jeanconn fixed the typo.